### ISSUE TITLE
Cluster Deletion Should Say Cluster Doesn't Exist When Deleting Not Existent Cluster

### ISSUE THREAD

**Issue by danielhelfand** [kubernetes-sigs/kind#1205]
**What happened**:

Running `kind delete cluster --name` for a cluster that doesn't exist returns the following:

```
Deleting cluster "notexist" ...
```

**What you expected to happen**:

A message saying something along the lines that the cluster doesn't exist. Otherwise, it makes it seem like the deletion occurred without issue. 

**How to reproduce it (as minimally and precisely as possible)**:

Run `kind delete cluster --name notexist` 

**Anything else we need to know?**:

It would also be nice to put some kind of confirmation message after a cluster has been deleted.

**Environment:**

- kind version: (use `kind version`): master branch

**Comment by BenTheElder** [comment]
Clusters don't really exist, kind doesn't have it's own state store.

Kind can be aware of no nodes, but it will still cleanup kubeconfig (removing the stale context)

**Comment by danielhelfand** [comment]
This could be accomplished by pulling back the list of clusters and checking if the specified name is present or, if there are no nodes found, returning an error message stating that no nodes are present.

**Comment by BenTheElder** [comment]
A bit more detail...

`kind delete cluster` is indempotent / re-entrant, and includes _all_ cluster cleanup, including e.g. kubeconfig entries...

kind does not itself have a concept of a cluster existing or not, except to help users with `kind get clusters`.

commands like `kind create clusters` have moved to clearer messaging on this to note that _nodes_ exist for a cluster already.

we could maybe post a warning, but I think we're giving the wrong signal if we do this, users should 100% call `kind delete cluster --name=foo` regardless of `foo` existing, e.g. like:

```shell
#!/bin/sh

name="foo"
cleanup() {
  kind delete cluster --name="${name}"
}
trap cleanup EXIT

kind create cluster --name="${name}"

# do tests
testy_test

# cleanup
kind delete cluster

# do post test things
green_eggs_and_ham
```

this is actually the preferred pattern

**Comment by BenTheElder** [comment]
TLDR I think this sends the wrong message that deleting a cluster if it "doesn't exist" is wrong when it's actually perfectly fine and a desired method of cleanup.

**Comment by danielhelfand** [comment]
I think this would be nice to have even if the cluster doesn't truly exist, but I realize that this would be complicated for not much gain. I'll close this for now and thanks for explaining.
